### DISCUSSION TITLE
cookies not available in first visit to url with locale suffix

### DISCUSSION THREAD

**Discussion by imthegrv** [nuxt-modules/i18n#1369] [General]
I am working on a travel application using nuxt and using i18n-module for language locale. i am facing a problem here.
I want to get cookie value in fetch or asyncdata on page level.

like this, i store some cookies in global middleware and get that cookie in asyncdata hook for api parameters. i am trying to print here.

`async asyncData({req}) {
        console.log(req.headers.cookie)
    },`

here the problem is.
after clearing cookie, in fresh browser, if i visit localhost:3000 in browser, it redirects me to localhost:3000/en-IN and i can see all stored cookies are getting printed...
but if in fresh browser, i directly visit localhost:3000/en-IN ,, it prints cookie undefined...
it is happening only when i am visiting the url for the first time using locale, after first run, there will be no issue at all.

looking for some suggestions.

**Comment by rchl** [comment]
This is a normal browser behavior. `req.headers.cookie` represents the cookies that the browser has sent and there aren't any cookies sent on the first request.

How to handle your use case depends on what exactly are you trying to do but this is not really a good place to discuss that. I'd suggest more generic forum like https://discord.com/invite/ps2h6QT if you need more help.
